src/agent/application.hpp

```diff
@@ -105,7 +109,13 @@ class Application
 
     static void HandleSignal(int aSignal);
 
+    Ncp::ControllerOpenThread &mPlaceHolder;
```

Comment:
What is the purpose of `mPlaceHolder`?

Reply:
All members/features in class `Application` is compile-time optional, we need to put something behind the member initializer list symbol (i.e. `:`) to get it compile:
```c++
Application::Application(Ncp::ControllerOpenThread &aOpenThread)
    : mPlaceHolder(aOpenThread)
#if OTBR_ENABLE_BORDER_AGENT
    , mBorderAgent(aOpenThread)
#endif
#if OTBR_ENABLE_BACKBONE_ROUTER
    , mBackboneAgent(aOpenThread)
#endif
#if OTBR_ENABLE_OPENWRT
    , mUbusAgent(aOpenThread)
#endif
#if OTBR_ENABLE_REST_SERVER
    , mRestWebServer(aOpenThread)
#endif
#if OTBR_ENABLE_DBUS_SERVER
    , mDBusAgent(aOpenThread)
#endif
#if OTBR_ENABLE_VENDOR_SERVER
    , mVendorServer(aOpenThread)
#endif
{
    OTBR_UNUSED_VARIABLE(mPlaceHolder);
}
```

Any suggestions to get it work without an additional member?


// @simonlingoogle 
// @bukepo

Reply:
Make Application into a singleton and change `sShouldTerminate` to non-static?

Reply:
> Make Application into a singleton and change `sShouldTerminate` to non-static?


It may not be preferred because:
1. It requires more changes;
2. Should avoid creating singleton if it doesn't improve readability or performance - singleton lifecycle management needs more attentions.

Reply:
`sShouldTerminate` needs to be access in signal handler. A singleton seems to be safer.

Reply:
> `sShouldTerminate` needs to be access in signal handler. A singleton seems to be safer.


`sShouldTerminate` is a static field already.